xpdacq/beamtime.py

```diff
@@ -1,720 +1,408 @@
-        ftype = self.type
-        fname = self._name_for_obj_yaml_file(oname,ftype)
-        fpath = os.path.join(self._yaml_path(), fname)
-        objlist = _get_yaml_list()
-        objlist = _update_objlist(objlist, fname)
-        with open(lname, 'w') as fout:
-            yaml.dump(objlist, fout)
-
-        if isinstance(fpath, str):
-            with open(fpath, 'w') as fout:
-                yaml.dump(self, fout)
+import inspect
+from mock import MagicMock
+from collections import ChainMap
+import bluesky.plans as bp
+import numpy as np
+from bluesky import RunEngine
+from bluesky.utils import normalize_subs_input
+from bluesky.callbacks import LiveTable
+#from bluesky.callbacks.broker import verify_files_saved
+
+from .glbl import glbl
+from .yamldict import YamlDict, YamlChainMap
+from .validated_dict import ValidatedDictLike
+
+
+# This is used to map plan names (strings in the YAML file) to actual
+# plan functions in Python.
+_PLAN_REGISTRY = {}
+
+def register_plan(plan_name, plan_func, overwrite=False):
+    "Map between a plan_name (string) and a plan_func (generator function)."
+    if plan_name in _PLAN_REGISTRY and not overwrite:
+        raise KeyError("A plan is already registered by this name. Use "
+                       "overwrite=True to overwrite it.")
+    _PLAN_REGISTRY[plan_name] = plan_func
+
+
+def unregister_plan(plan_name):
+    del _PLAN_REGISTRY[plan_name]
+
+
+def _summarize(plan):
+    "based on bluesky.utils.print_summary"
+    output = []
+    read_cache = []
+    for msg in plan:
+        cmd = msg.command
+        if cmd == 'open_run':
+            output.append('{:=^80}'.format(' Open Run '))
+        elif cmd == 'close_run':
+            output.append('{:=^80}'.format(' Close Run '))
+        elif cmd == 'set':
+            output.append('{motor.name} -> {args[0]}'.format(motor=msg.obj,
+                                                             args=msg.args))
+        elif cmd == 'create':
+            pass
+        elif cmd == 'read':
+            read_cache.append(msg.obj.name)
+        elif cmd == 'save':
+            output.append('  Read {}'.format(read_cache))
+            read_cache = []
+    return '\n'.join(output)
+
+
+def _configure_pe1c(exposure):
+    """ priviate function to configure pe1c with continuous acquistion
+    mode"""
+    #cs studio configuration doesn't propagate to python level
+    glbl.area_det.cam.acquire_time.put(glbl.frame_acq_time)
+    acq_time = glbl.area_det.cam.acquire_time.get()
+    # compute number of frames
+    num_frame = np.ceil(exposure / acq_time)
+    if num_frame == 0:
+        num_frame = 1
+    computed_exposure = num_frame*acq_time
+    glbl.area_det.images_per_set.put(num_frame)
+    # print exposure time
+    print("INFO: requested exposure time = {} - > computed exposure time"
+          "= {}".format(exposure, computed_exposure))
+    return (num_frame, acq_time, computed_exposure)
+
+
+def ct(dets, exposure, *, md=None):
+    pe1c, = dets
```

Comment:
When you get around to that (and I agree it's reasonable to do it in a separate pass, given the overhaul nature of this PR) you can save time by copy/pasting the bluesky count docstring (etc.) and editing as needed.